src/components/navigation/navigation.tsx

```diff
@@ -0,0 +1,201 @@
+const subItem4SVG = require('@/assets/images/sub-item4.svg');
+
+import { cn } from '@/lib/utils';
+
+const navigationVariants = cva(
+  'inline-flex justify-between w-full p-3 rounded-lg cursor-pointer text-sm font-normal',
+  {
+    variants: {
+      variant: {
+        highlighted:
+          'shadow shadow-[0px_0px_0px_3px_rgba(14,159,110,0.12)] hover:bg-list-highlighted text-success-foreground active:font-semibold',
+        default:
+          'text-tertiary-foreground hover:text-active-foreground hover:bg-list-actived active:font-semibold active:bg-active-menu active:text-active-foreground',
+        disabled: 'text-tertiary-foreground opacity-45 cursor-not-allowed',
+      },
+    },
+    defaultVariants: {
+      variant: 'default',
+    },
+  }
+);
+
+export interface NavigationLinkProps
+  extends React.ComponentProps<typeof NavigationMenuPrimitive.Link>,
+    VariantProps<typeof navigationVariants> {}
+
+function NavigationLink({ className, variant, ...props }: NavigationLinkProps) {
+  return (
+    <NavigationMenuPrimitive.Root>
+      <NavigationMenuPrimitive.Link
+        data-testid='navigation-link'
+        className={cn(navigationVariants({ variant }), className)}
+        {...props}
+      />
+    </NavigationMenuPrimitive.Root>
+  );
+}
+
+export interface NavigationItemProps
+  extends React.ComponentProps<typeof NavigationMenuPrimitive.Item>,
+    VariantProps<typeof navigationVariants> {}
+
+function NavigationItem({ className, variant, ...props }: NavigationItemProps) {
+  const [isActive, setIsActive] = React.useState(false);
+
+  return (
+    <NavigationMenuPrimitive.Root>
+      <NavigationMenuPrimitive.Item
+        data-testid='navigation-item'
+        className={cn(
+          navigationVariants({ variant }),
+          className,
+          isActive &&
+            !variant &&
+            'text-active-foreground bg-active-menu font-semibold',
+          isActive &&
+            variant === 'highlighted' &&
+            'text-success-foreground bg-list-highlighted font-semibold'
+        )}
+        onClick={() => setIsActive(!isActive)}
+        {...props}
+      />
+    </NavigationMenuPrimitive.Root>
+  );
+}
+
+export interface NavigationSubItemProps
+  extends React.ComponentProps<typeof NavigationMenuPrimitive.Link> {
+  position?: 'first' | 'middle' | 'last' | 'only';
+}
+
+function NavigationSubItem({
+  className,
+  position,
+  ...props
+}: NavigationSubItemProps) {
+  const handleIcon = (position: string) => {
+    if (position === 'first') {
+      return <Image src={subItem1SVG} alt='dot-icon' data-testid='first-dot' />;
+    } else if (position === 'last') {
+      return <Image src={subItem3SVG} alt='dot-icon' data-testid='last-dot' />;
+    } else if (position === 'only') {
+      return <Image src={subItem4SVG} alt='dot-icon' data-testid='only-dot' />;
+    } else
+      return <Image src={subItem2SVG} alt='dot-icon' data-testid='middle' />;
+  };
```

Comment:
You can remove the `else if`, since you're using `return`
```suggestion
  const handleIcon = (position: string) => {
    if (position === 'first') {
      return <Image src={subItem1SVG} alt='dot-icon' data-testid='first-dot' />;
    }
    if (position === 'last') {
      return <Image src={subItem3SVG} alt='dot-icon' data-testid='last-dot' />;
    } 
    if (position === 'only') {
      return <Image src={subItem4SVG} alt='dot-icon' data-testid='only-dot' />;
    } 
    
    return <Image src={subItem2SVG} alt='dot-icon' data-testid='middle' />;
  };
```

Besides that, we can change this function name to contain `render` instead of `handle`. Something like `renderLeftIconPosition`